src/pybind/mgr/cephadm/module.py

```diff
@@ -1632,13 +1632,16 @@ def _create_daemon(self, daemon_type, daemon_id, host,
         if extra_config:
             config += extra_config
 
-        # crash_keyring
-        ret, crash_keyring, err = self.mon_command({
-            'prefix': 'auth get-or-create',
-            'entity': 'client.crash.%s' % host,
-            'caps': ['mon', 'profile crash',
-                     'mgr', 'profile crash'],
-        })
+        if daemon_type != 'crash':
```

Comment:
```suggestion
        if daemon_type == 'crash':
```

Reply:
This is correct.. we only need to specify the crash keyring alongside non-crash daemons (which implicitly deploy both the regular daemon with it's ``keyring`` and the crash container with ``crash_keyring``).

Reply:
ah, got it. :+1: